internal/cloud_provider/resource_cloud_provider.go

```diff
@@ -329,6 +350,12 @@ func buildConfig(d *schema.ResourceData) (map[string]interface{}, error) {
 				config["network"] = network
 			}
 		}
+		if !isIAM {
+			config, err = utils.GcpGetCredentialsAsMap()
```

Comment:
This assignment would remove the previous entries from the `config` map? Could you please check this

Reply:
Fixed